File: src/main/java/dev/elysium/eraces/events/PlayerQuitListener.java
package dev.elysium.eraces.events;

import dev.elysium.eraces.RacesReloader;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerQuitEvent;

public class PlayerQuitListener implements Listener {
    @EventHandler
    public void onPlayerQuit(PlayerQuitEvent event) {
        RacesReloader.unloadPlayerData(event.getPlayer());
    }
}

File: src/main/java/dev/elysium/eraces/listeners/PlayerQuitListener.kt
package dev.elysium.eraces.listeners

import dev.elysium.eraces.RacesReloader.onPlayerLeave
import dev.elysium.eraces.VisualsManager.unloadVisualsForPlayer
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerQuitEvent

class PlayerQuitListener : Listener {
    @EventHandler
    fun onPlayerQuit(event: PlayerQuitEvent) {
        onPlayerLeave(event.player)
        unloadVisualsForPlayer(event.player)
    }
}
